### ISSUE TITLE
No GUI way to set or upload backgrounds

### ISSUE THREAD

**Issue by s2156945** [meerkat-dashboard/meerkat#142]
**Describe the bug**
It is not obvious/possible to upload a background or set its location.  Existing backgrounds (e.g. dashboards ported from Meerkat 2) do work.

**To Reproduce**
1.create a new dashboard
2. Try and upload a background

**Comment by ollytom** [comment]
To ease error handling and a bunch of state bugs in the forms in the editor panel I created a separate dashboard info page.
The info page has no javascript.

You can get to it by clicking the dashboard title in the home page.
Do you think this is clear? Maybe a button next to "Edit" and "View", e.g. "More info"?

Anyway from that page we can specify a link to an image.

I've deliberately left image uploading for later as I figured
it more web-like to just link to a file rather than needing to copy bytes to another disk
when it probably was stored on another server already.

There were other issues needing to be addressed re: uploading files
(https://github.com/meerkat-dashboard/meerkat/issues/130,
https://github.com/meerkat-dashboard/meerkat/issues/130
https://github.com/meerkat-dashboard/meerkat/issues/93
https://github.com/meerkat-dashboard/meerkat/issues/25).

Also:

* access control
* if running in ephemeral environment e.g. kubernetes, docker, whatever, did we remember to bind mount the background image directory?
* backups
* giving a human-readable name to files

Long story short when the state of the system was so bad that basic things like checking HTTP response codes was not being done
I couldn't think about doing a safe file upload feature!

I'm really happy that we can now focus on these things rather than all that stuff from before!!

**Comment by s2156945** [comment]
I think giving them what they are used to is a good solutions:
- we need to support it for ported dashboards
- it means no extra firewall rules to get images from wherever

**Comment by sol1-matt** [comment]
> I've deliberately left image uploading for later as I figured it more web-like to just link to a file rather than needing to copy bytes to another disk when it probably was stored on another server already.

This is a incorrect assumption

* 99% of backgrounds are loaded from disk, they don't come from existing online sources
* backgrounds for dashboards are changed often when creating new dashboards the the relationship between the background and elements is changed or tuned

**Comment by sol1-matt** [comment]
restored file upload in backgrounds in ec86583
